### PR TITLE
[BugFix][310P] fix performance degrade issue with MTP for Qwen3.5-2B on the 310P

### PR DESCRIPTION
**Pull request by Thiagor2002**
### What this PR does / why we need it?

On Ascend **310P**, enabling MTP (`num_speculative_tokens=1`) for **Qwen3.5-2B-W8A8** can make ais_bench synthetic decode **slower** than MTP-off (higher TPOT / lower output throughput), while larger models such as Qwen3.5-9B still benefit from MTP on the same stack.

**Root cause:** MTP=1 does ≥2 FULL ACLGraph replays per step (target SpecDecoding verify + draft). On 310P SpecDecoding, capture-stable buffer refreshes already run on the **compute stream**, but the path still paid heavy **device-wide `synchronize()`** barriers, plus **update↔compute handshakes even when `UpdatableGraph` has no `graph_task`s** (310P attention typically does not register FIA/PA tasks), plus repeated **D2H / `.item()`** on rejection and draft-prefill metadata. Short 2B kernels make that fixed overhead dominate; 9B compute can hide it.

**What this PR changes:**
1. **Weaken / remove hard sync before FULL replay**
   - `ACLGraphWrapper`: `synchronize()` → `wait_stream(update_stream)`; skip wait when the updatable graph has no tasks; skip empty `graph.update`.
   - Remove speculative `synchronize()` in 310P MRV2 target/draft ACLGraph managers.
   - MRV1 (`NPUModelRunner310`): skip runner-side `update_before_replay` barriers when `use_updatable_graph` (ordering owned by `ACLGraphWrapper`).
2. **Skip empty update↔compute handshake** in MRV2 `ModelAclGraphManager` / draft `AutoRegressiveAclGraphManager` when `resolved_tasks` is empty (no-op for backends that still register FIA/PA tasks).
3. **Host mirrors to cut MTP hot-path D2H** on 310P MRV2: reuse rejection CPU buffers / `set_prefill_host_meta` for draft-prefill; prefer host `query_lens` / `seq_lens_np` / `num_actual_tokens` in attention mask & prefill; avoid `current_draft_step.item()` via `_DRAFT_STEP_HOST`.
4. **Startup fix:** draft `ModelConfig.hf_overrides` may be a **callable** from `SpeculativeConfig.compose_draft_hf_overrides`. Ascend W8A8 has no HF `quantization_config`, so `VllmConfig` rebuild → `get_quant_config` requires dict overrides. Copy draft config and normalize non-dict overrides to `{}` before `replace()` (do not mutate the shared speculative config).
Does **not** change MTP accept/reject semantics or capture-size contract (`cudagraph_capture_sizes` must remain multiples of `(1+K)×batch`).

### Does this PR introduce _any_ user-facing change?

- **Behavior:** No API change. MTP accuracy path unchanged.
- **Perf:** 310P small hybrid models (e.g. Qwen3.5-2B-W8A8) with MTP should no longer regress vs MTP-off / should meet the reference numbers below under the stated bench.
- **Ops:** Recommend `--quantization ascend` (or rely on auto-detect) and, for random synthetic ais_bench, `--no-enable-prefix-caching` to stabilize TTFT (prefix hits are rare on random tokenids).

### How was this patch tested?

Performance testing data on 310P with `MTP=1` on MRV2/V1 for Qwen3.5-2B

command for vllm seve:
```bash
vllm serve /home/weights/Qwen3.5-2B-W8A8/ \
--trust-remote-code --dtype float16 \
--max-model-len 16384 \
--gpu_memory_utilization 0.9 \
--max-num-seqs 32 \
--mamba-ssm-cache-dtype float16   \
--quantization ascend  --allowed-local-media-path /  \
--compilation-config '{"cudagraph_mode": "FULL_DECODE_ONLY", "cudagraph_capture_sizes": [2,4,8,16,32]}'  \
--additional-config '{"ascend_compilation_config":{"fuse_norm_quant": false,"enable_npugraph_ex":false}}'  \
--host 127.0.0.1 --port 2026  \
--served-model-name qwen3.5 \
--speculative-config '{"method": "mtp","num_speculative_tokens": 1}' \
--no-enable-prefix-caching
```
preformance data of Qwen3.5-2B-w8a8

| Config | TP | Input-Output size | item | BS | TTFT | TPOT | Out tok/s | Total tok/s |
|--------|------|------|-----------|-------------|------|------|-----------|-------------|
| Before MRV2 noMTP| 1 | 1024-1024 | 4 | 1 | 626.7 | 21.7 | 44.03 | 120.71 |
| Before MRV2 MTP| 1 | 1024-1024 | 4 | 1 | 644.4 | 25.9 | 38.885 | 124.31 |
| Before MRV1 MTP| 1 | 1024-1024 | 4 | 1 | 651.9 | 23.8 |  41.12 |  94.53 |
| After MRV2 MTP |  1 | 1024 -1024| 4 | 1 | 603.6 | 21.8 | 44.78 | 93.48 |
| After MRV1 MTP |   1 | 1024 -1024| 4 | 1 | 578.3 | 20.4 | 47.74 | 99.96 |

After MRV1 with MTP
image

After MRV2 with MTP
image

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

**Comment by Thiagor2002** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

File: vllm_ascend/_310p/worker/v2/spec_decode/mtp_speculator.py [hunk 1]
```diff
@@ -158,8 +167,13 @@ def propose(self, *args, **kwargs):  # type: ignore[no-untyped-def]
...
-        if isinstance(num_rejected, torch.Tensor):
-            self._last_num_rejected_cpu = num_rejected.detach().to("cpu")
+        # Prefer host mirror published by postprocess_sampled (already synced).
+        # Avoid an extra device→host copy on every MTP propose.
+        if getattr(self, "_last_num_rejected_cpu", None) is None and isinstance(num_rejected, torch.Tensor):
+            if num_rejected.device.type == "cpu":
+                self._last_num_rejected_cpu = num_rejected
+            else:
+                self._last_num_rejected_cpu = num_rejected.detach().to("cpu")
         return super().propose(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (Thiagor2002) [reply vllm_ascend/_310p/worker/v2/spec_decode/mtp_speculator.py:177]: Agreed — adopted. propose() now wraps super().propose in try/finally and sets _last_num_rejected_cpu = None after consume. Hot path stays cheap: each step’s postprocess_sampled republishes the CPU mirror before the next propose, so we still avoid the extra D2H. Clearing prevents a later propose without a fresh postprocess from reusing a stale rejection count when num_rejected is present but the mirror is still set.

File: vllm_ascend/_310p/worker/v2/spec_utils.py [hunk 2]
```diff
@@ -326,11 +328,19 @@ def prepare_prefill_inputs_cpu(
...
         seq_lens_np = input_batch.seq_lens.detach().cpu().numpy()
 
-    # Small metadata only (one sync each); avoid D2H of full token buffers.
-    num_sampled_np = num_sampled[:num_reqs].detach().cpu().numpy()
-    num_rejected_np = num_rejected[:num_reqs].detach().cpu().numpy()
-    last_sampled_np = last_sampled.detach().cpu().numpy()
-    next_prefill_np = next_prefill_tokens.detach().cpu().numpy()
+    # Prefer host mirrors published after rejection synchronize_cpu (no D2H).
+    host_meta = _PREFILL_HOST_META
+    if host_meta is not None and int(host_meta.get("num_reqs", -1)) == num_reqs:
```
[the remark this answers is not in the document]
Comment (Thiagor2002) [reply vllm_ascend/_310p/worker/v2/spec_utils.py:333]: We intentionally do not clear `_PREFILL_HOST_META` on consume. `postprocess_sampled` overwrites the mirror every speculative step via `set_prefill_host_meta(...)` after `synchronize_cpu()`. Clearing after `prepare_prefill_inputs_cpu` would reintroduce the `4× D2H` we removed on 310P (and we previously deleted unused `clear_prefill_host_meta` for the same reason). The `num_reqs` match is a cheap guard; freshness is guaranteed by `republish-on-postprocess`, not by consume-once clearing.

**Comment by Thiagor2002** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E
